### DISCUSSION TITLE
[Feature Request] - Calculate dynamic layout dimensions

### DISCUSSION THREAD

**Discussion by kevin-kessler** [rxhanson/RectanglePro-Community#608] [Ideas]
Hey there!
First of all: thanks for this awesome piece of software - I love it! 

I'd like to suggest a feature concerning App Layout Dimensions.

### Suggestion
 I'd like to be able to define a percentage of the screen width/height and subtract a fix amount of pixels from it. Similar to what you can do with [CSS calc()](https://developer.mozilla.org/en-US/docs/Web/CSS/calc).
![image](https://github.com/user-attachments/assets/f535e332-b654-4cc6-a67d-a682bd33d701)

### Use case
I have a fixed size menu bar at the bottom of my screen. Putting a window at 100% height, will cause overlap with this menu bar. I'd like to be able to define 100% height and subtract the height of that menu bar (e.g. 50px).

### Example
![image](https://github.com/user-attachments/assets/066a991d-00a5-4d5a-8429-f45645eb7d01)

### Result
![image](https://github.com/user-attachments/assets/8c7bf21e-0412-4deb-bea2-a2cc57e3f07b)

Thanks for considering!

**Comment by rxhanson** [comment]
Thanks for the detailed idea. I have this on my backlog but had not thought of doing it in this way, allowing it in the input field. I like it and will have to try it out to see what the best path forward will be.

In the meantime, you can set a screen edge gap at the bottom with the following Terminal command:
```
defaults write com.knollsoft.Hookshot screenEdgeGapBottom -int 50
```
Then restart the app.

I also have plans to add screenEdgeGap settings to the UI and allow per-display configuration.

**Reply by kevin-kessler** [reply]
Thanks a lot for the quick response! Glad to hear it's already on your backlog. And thanks a lot for the workaround with the screen edge gap - works like a charm!
